migrator/src/main/java/com/redhat/lightblue/migrator/Controller.java

```diff
@@ -0,0 +1,121 @@
+package com.redhat.lightblue.migrator;
+
+import java.io.IOException;
+
+import java.util.Map;
+import java.util.HashMap;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.redhat.lightblue.client.LightblueClient;
+import com.redhat.lightblue.client.request.data.DataFindRequest;
+import com.redhat.lightblue.client.http.LightblueHttpClient;
+import com.redhat.lightblue.client.hystrix.LightblueHystrixClient;
+import com.redhat.lightblue.client.expression.query.ValueQuery;
+
+import static com.redhat.lightblue.client.expression.query.ValueQuery.withValue;
+import static com.redhat.lightblue.client.projection.FieldProjection.includeFieldRecursively;
+
+public class Controller extends Thread {
+
+    private static final Logger LOGGER=LoggerFactory.getLogger(Controller.class);
+    
+    private final MainConfiguration cfg;
+    private final LightblueClient lightblueClient;
+    private final Map<String,MigrationProcess> migrationMap=new HashMap<>();
+
+
+    public static class MigrationProcess {
+        public final MigrationConfiguration cfg;
+        public final MigratorController mig;
+
+        public MigrationProcess(MigrationConfiguration cfg,
+                                MigratorController mig) {
+            this.cfg=cfg;
+            this.mig=mig;
+        }
+    }
+    
+    public Controller(MainConfiguration cfg) {
+        this.cfg=cfg;
+        this.lightblueClient=getLightblueClient();
+    }
+
+    public Map<String,MigrationProcess> getMigrationProcesses() {
+        return migrationMap;
+    }
+
+    public MainConfiguration getMainConfiguration() {
+        return cfg;
+    }
+    
+    /**
+     * Read configurations from the database whose name matches this instance name
+     */
+    public MigrationConfiguration[] getMigrationConfigurations()
+        throws IOException {
+        DataFindRequest findRequest = new DataFindRequest("migrationConfiguration",null);
+        findRequest.where(withValue("consistencyCheckerName = " + cfg.getName()));
+        findRequest.select(includeFieldRecursively("*"));
+        LOGGER.debug("Loading configuration:{}",findRequest.getBody());
+        return lightblueClient.data(findRequest, MigrationConfiguration[].class);
+    }
+    
+    public LightblueClient getLightblueClient() {
+        LightblueHttpClient httpClient;
+        if (cfg.getClientConfig() != null) {
+            httpClient = new LightblueHttpClient(cfg.getClientConfig());
+        } else {
+            httpClient = new LightblueHttpClient();
+        }
+        return new LightblueHystrixClient(httpClient, "migrator", "primaryClient");
+    }
```

Comment:
Makes sense, i'll change

On Wed, Jun 3, 2015 at 8:06 AM, Dennis Crissman <notifications@github.com>
wrote:

> In migrator/src/main/java/com/redhat/lightblue/migrator/Controller.java
> <https://github.com/lightblue-platform/lightblue-migrator/pull/197#discussion_r31625556>
> :
>
> > +        DataFindRequest findRequest = new DataFindRequest("migrationConfiguration",null);
> > +        findRequest.where(withValue("consistencyCheckerName = " + cfg.getName()));
> > +        findRequest.select(includeFieldRecursively("*"));
> > +        LOGGER.debug("Loading configuration:{}",findRequest.getBody());
> > +        return lightblueClient.data(findRequest, MigrationConfiguration[].class);
> > +    }
> > +
> > +    public LightblueClient getLightblueClient() {
> > +        LightblueHttpClient httpClient;
> > +        if (cfg.getClientConfig() != null) {
> > +            httpClient = new LightblueHttpClient(cfg.getClientConfig());
> > +        } else {
> > +            httpClient = new LightblueHttpClient();
> > +        }
> > +        return new LightblueHystrixClient(httpClient, "migrator", "primaryClient");
> > +    }
>
> We could call it createLightblueClient, but I wouldn't block because of
> that.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/lightblue-platform/lightblue-migrator/pull/197/files#r31625556>
> .
>